### PR TITLE
PIRK-29

### PR DESCRIPTION
**Pull request by jkovba**
Implemented a check to ensure the queryID (formerly named queryName) in the Querier matches the queryID in the Response object.

Added a timestamp to the queryID for uniqueness.

Renamed some fields/methods for uniformity.

### CODE REVIEW

File: src/main/java/org/apache/pirk/querier/wideskies/QuerierDriverCLI.java [hunk 1]
```diff
@@ -239,12 +242,19 @@ private boolean parseOptions()
...
-        logger.info("Must have the option " + QUERYNAME);
+        logger.info("Must have the option " + QUERYID);
         return false;
       }
-      SystemConfiguration.setProperty(QUERYNAME, getOptionValue(QUERYNAME));
+      
+      // create a formatted date/time string
+      Date date = new Date();
+      SimpleDateFormat sdf = new SimpleDateFormat("MM.dd.yyyy_hh.mm.ss");
+      String formattedDate = sdf.format(date);
```
Comment (smarthi) [inline src/main/java/org/apache/pirk/querier/wideskies/QuerierDriverCLI.java]: Could we use Java 8 LocalDateTime and DateTimFormatter APIs instead ?

File: src/main/java/org/apache/pirk/querier/wideskies/QuerierDriver.java [hunk 2]
```diff
@@ -91,7 +91,7 @@ public static void main(String... args) throws IOException, InterruptedException
     int dataPartitionBitSize = 0;
     int paillierBitSize = 0;
     int certainty = 0;
-    String queryName = null;
```
Comment (ellisonanne) [inline src/main/java/org/apache/pirk/querier/wideskies/QuerierDriver.java]: The queryName corresponds to the schemaName in the query-schema.xsd file *not* to the user-given identifier for the  query -- the identifier for this query is actually the queryNum. Notice that queryNum is currently a double -- if we want to start appending dates or whatever else to it, we need to change its type to String throughout.

File: src/main/java/org/apache/pirk/querier/wideskies/QuerierDriverCLI.java [hunk 3]
```diff
@@ -55,7 +58,7 @@
   public static final String PAILLIERBITSIZE = "paillierBitSize";
   public static final String BITSET = "bitSet";
   public static final String CERTAINTY = "certainty";
-  public static final String QUERYNAME = "queryName";
+  public static final String QUERYID = "queryID";
```
Comment (ellisonanne) [inline src/main/java/org/apache/pirk/querier/wideskies/QuerierDriverCLI.java]: See previous comment on the QuerierDriver:

The queryName corresponds to the schemaName in the query-schema.xsd file not to the user-given identifier for the query -- the identifier for this query is actually the queryNum. Notice that queryNum is currently a double -- if we want to start appending dates or whatever else to it, we need to change its type to String throughout.

File: src/main/java/org/apache/pirk/query/wideskies/QueryInfo.java [hunk 4]
```diff
@@ -43,7 +43,7 @@
 
   private String queryType = null; // QueryType string const
 
-  private String queryName = null; // Name of query
+  private String queryID = null; // Unique query ID
```
Comment (ellisonanne) [inline src/main/java/org/apache/pirk/query/wideskies/QueryInfo.java]: See previous notes above

**Comment by ellisonanne** [comment]
I think that these changes look good. 

Joe - Can you please merge this PR branch with the current master (your base version of master is rather 'old')? Then, it will be easy to accept and merge the pull request. Thanks!

File: src/main/java/org/apache/pirk/querier/wideskies/QuerierDriverCLI.java (hunk 1 above)
Comment (DarinJ) [inline src/main/java/org/apache/pirk/querier/wideskies/QuerierDriverCLI.java]: According to: https://pirk.incubator.apache.org/for_users, the required JDK is 1.7.  I'm pro 1.8, but that's not the subject of this PR.  Was the decision made to drop 1.7 support?

Comment (smarthi) [inline src/main/java/org/apache/pirk/querier/wideskies/QuerierDriverCLI.java]: I believe we decided to go with Java 8 for Pirk and the POM has been updated to reflect that.  The link u point to needs to be updated too. Pirk is officially now on Java 8. Suggest u rebase ur  code with the present master and update this PR.

Comment (ellisonanne) [inline src/main/java/org/apache/pirk/querier/wideskies/QuerierDriverCLI.java]: Correct - we just went to 1.8 - I will update the website.

**Comment by ellisonanne** [comment]
I'm going to close this PR as the decision to make the query identifier to a UUID supersedes these changes.

**Comment by ellisonanne** [comment]
...Should also add that we can add in the line to perform the query ID check in the Querier (as in the corresponding JIRA issue) to a new PR.

**Pull request closed without merging**